scripts/deploy.sh

```diff
@@ -0,0 +1,136 @@
+    echo
+}
+
+addValuesToFile() {
+    result=$(jq -s add "$1" /dev/stdin)
+    printf %s "$result" > "$1"
+}
+
+deploy() {
+    echo -e "Deploying $1\n"
+    DEPLOYED_ADDR=$(dapp create "$2" "${@:3}")
+    verify "$2" "$DEPLOYED_ADDR" "${@:3}"
+    echo -e "\nDeployed $1 to $DEPLOYED_ADDR\n"
+}
+
+verify() {
+  if [ -n "$ETHERSCAN_API_KEY" ]
+  then
+      sleep 5 # give etherscan some time to process the block
+      dapp verify-contract --async "$@"
+  else
+      echo "ETHERSCAN_API_KEY not provided, skipping verification"
+  fi
+}
+
+# Set up the defaults
+DEPLOYMENT_JSON=${DEPLOYMENT_JSON:-./deployment_$(seth chain).json}
+RESERVE_OWNER=${RESERVE_OWNER:-$ETH_FROM}
+DRIPS_HUB_ADMIN=${DRIPS_HUB_ADMIN:-$ETH_FROM}
+CYCLE_SECS=${CYCLE_SECS:-$(( 7 * 24 * 60 * 60 ))} # 1 week
+
+# Print the configuration
+message Deployment Config
+echo "Network:                  $(seth chain)"
+echo "Deployer address:         $ETH_FROM"
+echo "Gas price:                ${ETH_GAS_PRICE:-use the default}"
+ETHERSCAN_API_KEY_PROVIDED=${ETHERSCAN_API_KEY:+provided}
+echo "Etherscan API key:        ${ETHERSCAN_API_KEY_PROVIDED:-not provided}"
+echo "Deployment JSON:          $DEPLOYMENT_JSON"
+TO_DEPLOY="to be deployed"
+echo "Reserve:                  ${RESERVE:-$TO_DEPLOY}"
+echo "Reserve owner:            $RESERVE_OWNER"
+echo "DripsHub:                 ${DRIPS_HUB:-$TO_DEPLOY}"
+echo "DripsHub admin:           $DRIPS_HUB_ADMIN"
+echo "DripsHub logic:           ${DRIPS_HUB_LOGIC:-$TO_DEPLOY}"
+echo "DripsHub cycle seconds:   $CYCLE_SECS"
+echo "AddressApp:               ${ADDRESS_APP:-$TO_DEPLOY}"
+echo
+
+read -p "Ready to deploy? [y/n] " -n 1 -r
+echo
+if [[ $REPLY =~ ^[^Yy] ]]
+then
+    exit 1
+fi
+
+# Build the contracts
+message Building Contracts
+dapp build
+
+# Deploy the contracts
+message Deploying contracts
+
+if [ -z "$RESERVE" ]; then
+    deploy "Reserve" 'src/Reserve.sol:Reserve' "$ETH_FROM"
+    RESERVE=$DEPLOYED_ADDR
+fi
+
+if [ -z "$DRIPS_HUB_LOGIC" ]; then
+    deploy "DripsHub logic" 'src/DripsHub.sol:DripsHub' "$CYCLE_SECS" "$RESERVE"
+    DRIPS_HUB_LOGIC=$DEPLOYED_ADDR
+fi
+
+if [ -z "$DRIPS_HUB" ]; then
+    deploy "DripsHub" 'src/Managed.sol:Proxy' "$DRIPS_HUB_LOGIC" "$DRIPS_HUB_ADMIN"
+    DRIPS_HUB=$DEPLOYED_ADDR
+fi
+
+if [ -z "$ADDRESS_APP" ]; then
+    deploy "AddressApp" 'src/AddressApp.sol:AddressApp' "$DRIPS_HUB"
+    ADDRESS_APP=$DEPLOYED_ADDR
+fi
+
+# Configuring the contracts
+if [ $(seth call "$RESERVE" 'isUser(address)(bool)' "$DRIPS_HUB") = "false" ]; then
+    echo -e "Adding DripsHub as a Reserve user\n"
+    seth send "$RESERVE" 'addUser(address)()' "$DRIPS_HUB"
+    echo
+fi
+
+if [ $(seth call "$RESERVE" 'owner()(address)') != $(seth --to-address $RESERVE_OWNER) ]; then
+    echo -e "Setting Reserve owner to $RESERVE_OWNER\n"
+    seth send "$RESERVE" 'transferOwnership(address)()' "$RESERVE_OWNER"
+    echo
+fi
+
+if [ $(seth call "$DRIPS_HUB" 'admin()(address)') != $(seth --to-address $DRIPS_HUB_ADMIN) ]; then
```

Comment:
I think this can never be false.

Because in the deployment of DRIPS_HUB you defined the admin already as `DRIPS_HUB_ADMIN`

Reply:
It can, if the user provides `DRIPS_HUB` themselves, it bypasses the deployment and setting to `DRIPS_HUB_ADMIN`. It's an edge case, but I think that it makes the script a little easier to handle by guaranteeing to always deliver what the user asks for.